lib/index_documents.rb

```diff
@@ -0,0 +1,86 @@
+require_relative "../app"
+require 'gds_api/publishing_api_v2'
+
+class IndexDocuments
+  def process(message)
+    raw_links = links_from_payload(message)
+
+    if !raw_links.nil? && !raw_links.empty?
+      links = remap_keys(raw_links, content_tagger_to_rummager_map)
+      links.each do |(link_type, content_ids)|
+        links[link_type] = paths_from_content_ids(content_ids)
+      end
+
+      base_path = document_base_path(message)
+      update_index(base_path, links)
+    end
+
+    message.ack
+  end
+
+private
+  def links_from_payload(message)
+    message.payload["links"].try(:slice, *content_tagger_tags)
+  end
+
+  def paths_from_content_ids(content_ids)
+    content_ids.map { |content_id| base_path content_id }.compact.sort
+  end
+
+  def content_tagger_tags
```

Comment:
Since the values here are required to be the keys for `content_tagger_to_rummager_map`, perhaps this would be better implemented by calling `.keys` on that, to avoid risk of them getting out of sync.